RUM-7809: Update SDK performance doc with Session Replay measurements (#2481)
DataDog/dd-sdk-android: docs/sdk_performance.md

[1]: https://github.com/Docile-Alligator/Infinity-For-Reddit
[2]: https://developer.android.com/studio/profile/jank-detection
[3]: https://developer.android.com/studio/profile/energy-profiler

> [4]: https://developer.android.com/topic/performance/vitals/launch-time#time-initial

Reviewer: I would recommend moving these reference links to the bottom of the page. Additionally, I'd switch the inline links added below to use the reference link format.

## Session Relay Performance Measurement

The Session Replay feature is expected to have a larger impact on both CPU and memory usage
within the application.

> This dedicated section provides detailed measurements, broken down by various scenarios, to offer a
> comprehensive analysis.

Reviewer: ```suggestion
This section provides detailed performance measurements across different scenarios.
```

### Methodology

> To simulate the typical usage of Datadog SDK it was added to
> the [Docile-Alligator/Infinity-For-Reddit][1] application and typical user behavior (scrolling the
> feed, browsing reddits) for Android view.
> 
> [NowInAndroid](https://github.com/android/nowinandroid) application is used for Jetpack Compose.
> 
> Each measurement last for 1 minute.
> 
> Application used: [Docile-Alligator/Infinity-For-Reddit][1] (
> revision [cfe1781](https://github.com/Docile-Alligator/Infinity-For-Reddit/tree/cfe1781))
> 
> [android/nowinandroid][1] (
> revision [cfe1781](https://github.com/android/nowinandroid/commit/904e6fccee809556b242898fb624f5f38200298c))
> 
> Device used: Samsung Galaxy S23
> Android OS: Android 14 (Build Number UP1A.231005.007)
> Datadog SDK: revision [2.17.0](https://github.com/DataDog/dd-sdk-android/tree/2.17.0)
> 
> CPU, Memory profiling was done using Android Studio Ladybug | 2024.2.1 Patch 2.
> Device had 3.7 GB memory free on average (out of 8 GB), 110 apps installed and 41.7 GB of storage
> was free.
> Device was connected to WiFi interface.
> 
> CPU and Memory profiling was done for the `debug` build type . CPU profiling was done using
> the `System Traces` option.

Reviewer: ```suggestion
To simulate the typical usage of the Datadog SDK with Session Replay enabled, measurements were performed using two applications:

Application used: [Docile-Alligator/Infinity-For-Reddit][1] (revision [cfe1781](https://github.com/Docile-Alligator/Infinity-For-Reddit/tree/cfe1781)) for Android View
[NowInAndroid](https://github.com/android/nowinandroid) (revision [904e6fccee809556b242898fb624f5f38200298c)) for Jetpack Compose

Device used: Samsung Galaxy S23
Android OS: Android 14 (Build Number UP1A.231005.007)
Datadog SDK: revision [2.17.0](https://github.com/DataDog/dd-sdk-android/tree/2.17.0)

CPU, Memory profiling was done using Android Studio Ladybug | 2024.2.1 Patch 2.
Device had 3.7 GB memory free on average (out of 8 GB), 110 apps installed and 41.7 GB of storage was free.
Device was connected to WiFi interface.

CPU and Memory profiling was done for the debug build type. CPU profiling was done using the System Traces option.

Each measurement lasted for 1 minute, during which typical user behavior was simulated (scrolling the feed, browsing content).

```

> ### Scenario corresponding

Reviewer: ```suggestion
### Scenario configuration
```
